### PR TITLE
feat(SHP-2062): variant metafields and number conversion

### PR DESCRIPTION
**Pull request by ezelopes-mohc**
- Updating go version
- Int conversion fix
- Including metafields against `variant` struct

### CODE REVIEW

File: metafields.go [hunk 1]
```diff
@@ -7,6 +7,27 @@ import (
...
+		converted, err := strconv.ParseInt(v.(string), 0, 64)
+		if err != nil {
+			fmt.Println("Error parsing string to int:", err)
+			return 0, err
+		}
+		return converted, nil
+	default:
+		return 0, fmt.Errorf("unsupported type %T", v)
+	}
+}
```
Comment (nichharp-mohc) [inline metafields.go:29]: Is there anywhere else that this might want to be used? Seems like a nice generic helper function.

**Review by nichharp-mohc — APPROVED** [review]
Looks good, 1 question

File: metafields.go (hunk 1 above)
Comment (ezelopes-mohc) [reply metafields.go:29]: It does, although it's only used in this file 🙂 if we require it somewhere else, we can move it into a helper function 👌

File: metafields.go [hunk 2]
```diff
@@ -7,6 +7,27 @@ import (
 	"time"
 )
 
+// ConvertToInt converts strings, float64 or int64 to int
```
Comment (shafliaquat) [inline metafields.go:10]:
```suggestion
// ConvertToInt64 converts strings, float64 or int to int64
```

**Review by shafliaquat — APPROVED** [review]
Lgtm, just that one comment that may need updating

**Pull request merged**